src/wasm/wasm-emscripten.cpp

```diff
@@ -47,15 +50,19 @@ void addExportedFunction(Module& wasm, Function* function) {
 }
 
 Global* EmscriptenGlueGenerator::getStackPointerGlobal() {
-  // Assumption: The first non-imported global is global is __stack_pointer
+  // Assumption: The stack pointer is either imported as __stack_pointer or
+  // its the first non-imported global.
   // TODO(sbc): Find a better way to discover the stack pointer.  Perhaps the
   // linker could export it by name?
   for (auto& g : wasm.globals) {
-    if (!g->imported()) {
+    if (g->imported()) {
+      if (g->base == "__stack_pointer")
```

Comment:
This one feels like it should have `{}`s on the `if`, due to the nesting.
(I'd thought binaryen style was to always have braces, grep says I'm wrong)